src/components/views/rooms/MessageContextMenu.js

```diff
@@ -57,25 +57,42 @@ module.exports = React.createClass({
         if (this.props.onFinished) this.props.onFinished();
     },
 
+    onCancelSendClick: function() {
+        Resend.removeFromQueue(this.props.mxEvent);
+        if (this.props.onFinished) this.props.onFinished();
```

Comment:
Yeah I thought about this but was like:
 - We can make no guarantees that they can cancel sending in time.
 - Adding in the glue to the JS-SDK guts to knife the HTTP connection is not my idea of a Fun Time, so screw that.

So was planning on basically putting my hands up and saying "we can make no guarantees! oooo!"